### PR TITLE
cuda archs for TX1 and TX2

### PR DESCRIPTION
**Pull request by dheera**
## Description ##
Add cuda archs for Jetson TX1 and TX2

Note also that one needs to add USE_F16C=0 to the make options to get this to work on Jetson. Suggest to add this to the MXnet installation documentation. Currently, people with TX1/TX2 would follow the given instructions, and are greeted with compile errors, and then spend a lot of time on StackOverflow. Here's the magic incantation version for TX1/TX2 with JetPack 3.2 (16.04) with CUDA 9.0, just leaving this on GitHub comments in hopes that people googling "mxnet jetson compile error" will find this comment:
make -j $(nproc) USE_OPENCV=1 USE_BLAS=openblas USE_CUDA=1 USE_CUDA_PATH=/usr/local/cuda USE_CUDNN=1 USE_F16C=0

That USE_F16C=0 needs to be done in addition to the cuda arch additions in this commit below.

(Suggestion: I wish all this F16C stuff were automatically detected and would "just work" ... especially on TX1/TX2, some of NVIDIA's most popular pieces of hardware ... people trying to work on ML shouldn't have to spend time dealing with installation problems. Ideally would be nice if it were officially supported to just be able to sudo pip3 install mxnet-cu90==1.2.0 on a TX2 and start deploying models within minutes).

### CODE REVIEW

**Comment by marcoabreu** [comment]
@KellenSunderland @aaronmarkham

**Comment by KellenSunderland** [comment]
Hey @dheera, good feedback on the installation process. We'll try and make it easier and then get the steps documented soon.

I'm not sure if you've seen this, but you can actually generate a pip installable wheel yourself by running the following command on an x86 machine:

```bash
./ci/build.py -p jetson
```

The only requirement is that it's run on a machine with Docker installed.  This will output a wheel that works on either a Jetson TX1 or TX2.   From that point just copy the wheel to your device and pip install it.

**Comment by aaronmarkham** [comment]
@KellenSunderland Does building the wheel in this way avoid having to use the following?
`make -j $(nproc) USE_OPENCV=1 USE_BLAS=openblas USE_CUDA=1 USE_CUDA_PATH=/usr/local/cuda USE_CUDNN=1 USE_F16C=0`
Are they different ways of getting to the same result? Any differences that would be important to mention?

**Comment by KellenSunderland** [comment]
@aaronmarkham Correct, you wouldn't have to run make manually if you use the ./ci/build.py script.  They are different ways of compiling the library that should be equivalent with one major difference.  You should run the ./ci/build.py script on an x86 machine, ideally one with plenty of cores to speed up compilation.  I wouldn't recommend compiling on device as each compile can take hours, and the build flags are artisanal.

**Comment by rahul003** [comment]
@KellenSunderland Would it make sense to update the install instructions here https://github.com/apache/incubator-mxnet/blob/master/docs/install/index.md to reference this config file https://github.com/apache/incubator-mxnet/blob/master/make/crosscompile.jetson.mk and the method you suggested? 
Are the instructions in that file (rendered here under Devices->Jetson https://mxnet.incubator.apache.org/install/index.html) up-to-date?

**Comment by KellenSunderland** [comment]
@rahul003 I think we should have a larger re-write of that section to include cross-compilation instructions, but you're right for the time being we should reference the config file that specifically targets Jetson devices.  Updated the docs in this PR: https://github.com/apache/incubator-mxnet/pull/10758/files

File: Makefile [hunk 1]
```diff
@@ -287,7 +287,7 @@ endif
 # be JIT-compiled by the updated driver from the included PTX.
 ifeq ($(USE_CUDA), 1)
 ifeq ($(CUDA_ARCH),)
-	KNOWN_CUDA_ARCHS := 30 35 50 52 60 61 70
+	KNOWN_CUDA_ARCHS := 30 35 50 52 53 60 61 62 70
```
Comment (szha) [inline Makefile]: could you make the KNOWN_CUDA_ARCHS configurable too?
`ifeq ($(KNOWN_CUDA_ARCHS),) then set the value`.

**Comment by lupesko** [comment]
@dheera - bouncing... are you planning to follow up on the comments and get this in shape for merge?
Thanks for the contribution!

**Comment by lupesko** [comment]
@dheera  - bouncing again.

**Comment by lupesko** [comment]
@szha @nswamy @sandeep-krishnamurthy kindly suggest to close the PR due to lack of response for > 1 month.

**Comment by sandeep-krishnamurthy** [comment]
Closing due to inactivity. Please reopen if closed in error

**Pull request closed without merging**